### ISSUE TITLE
terraform apply work fine and webhook configured but no instance running

### ISSUE THREAD

**Issue by salman-cheema** [philips-labs/terraform-aws-github-runner#1809]
I have created all the resources using this module and everything works fine but I do not see any instance running and my action remained in the Queue state because no instance(runner) is picking them up.

- Cloudwatch logs of lambda function `scale-up` show received event but no instance get register.

Is anyone else facing this issue?

**Comment by jokreliable** [comment]
Look in cloudwatch. There are a number of log groups for each component in the stack. Start with the time of the job submission and track the request from github through the webhook and the scale-up logs, as well as the runner-startup, user_data, messages, etc...
In there, you should find some indication as to why the runner didn't start and picked up the job.
you can also increase the debug level to get more detailed information that may help locate the problem. (see `log_level` [here](https://github.com/philips-labs/terraform-aws-github-runner#input_log_level) -- and scroll right on that table, it is quite a bit wider than it looks)

**Comment by npalm** [comment]
@jokreliable thx for the comment. In case we have an issue, we typically follow the follow the flow.

1. Check webhook in the github app (advanced section).
2. Check the log (lambda) of the webhook, result should be messaged added to the queue
3. Check the log (lambda) of the scale-up function, result should be instance created
4. When instances are created, logs of the instances should be in cloudwatch. If not you can enable SSM and jump in the instance and check the logs in `/var/log/`

**Comment by marcofranssen** [comment]
@jokreliable Facing the same issue on the `0.27.2` release. Trying to bump from `0.20.1`.

In cloudwatch userdata log I found the following.

```shell
2022-03-03T18:08:14.868+01:00 | + echo 'Starting the runner as user ec2-user'
-- | --
  | 2022-03-03T18:08:14.868+01:00 | Starting the runner as user ec2-user
  | 2022-03-03T18:08:14.868+01:00 | + [[ persistent = \e\p\h\e\m\e\r\a\l ]]
  | 2022-03-03T18:08:14.868+01:00 | + echo 'Installing the runner as a service'
  | 2022-03-03T18:08:14.868+01:00 | Installing the runner as a service
  | 2022-03-03T18:08:14.868+01:00 | + ./svc.sh install ec2-user
  | 2022-03-03T18:08:14.868+01:00 | /var/lib/cloud/instance/scripts/part-001: line 192: ./svc.sh: No such file or directory
  | 2022-03-03T18:08:14.868+01:00 | + echo 'Starting the runner in persistent mode'
  | 2022-03-03T18:08:14.868+01:00 | Starting the runner in persistent mode
  | 2022-03-03T18:08:14.868+01:00 | + ./svc.sh start
  | 2022-03-03T18:08:19.867+01:00 | /var/lib/cloud/instance/scripts/part-001: line 194: ./svc.sh: No such file or directory
```

Maybe you are facing the same issue. In the 0.20.1 release it still works however in the 0.27.2 one it is broken. Didn't try other releases in between yet. Also see #1810

**Comment by npalm** [comment]
@marcofranssen your issue is different, you got clearly a runner. The problem is that the script is using the ec2-user and seems you ar using ubuntu. So that won't work. Would suggest to upgrade to the latest version and have a look on the example of ubuntu and / or the image of the ubuntu for the start scripts.

**Comment by marcofranssen** [comment]
@npalm I successfully bumped to the v0.36.0 release.

Eventually there I had to do following fix in our user-data.sh vs `runner_run_as` variable.

Our historical user-data.sh template (from an example from roughly a year ago)

```diff
USER_NAME=runners
```

Took this line from the new example, but then had to align it with my version of the historical example.

```diff
- runner_run_as = "ubuntu"
+ runner_run_as = "runners"
```

See #1817 for a small improvement that resolves this for future module / example consumers.
